### PR TITLE
Add in progress computation

### PR DESCRIPTION
**Pull request by liadaram1**
Create a full flow to compute degree status with regard to courses in progress.

 - [x] add "settings" to User resource
 - [x] add compute_in_progress flag as a setting
 - [x] create PUT /student/settings
 - [x] before degree-status compute, check compute_in_progress flag and change courses statuses if needed.
 - [x] add toggle button UI to client
 - [x] change default course_status from 'complete' to 'in-progress'

### CODE REVIEW

**Review by benny-n — APPROVED** [review]

File: packages/sogrim-app/src/components/Banner/BannerCards/DegreeMainStatus.tsx [hunk 1]
```diff
@@ -1,60 +1,146 @@
-import { Button, Card, CardContent, Typography } from "@mui/material";
+import {
```
Comment (NissanOhana) [inline packages/sogrim-app/src/components/Banner/BannerCards/DegreeMainStatus.tsx:1]: we need to think about break out all the logic - and maybe create new sub-store for the `DegreeMainStatus`

File: packages/sogrim-app/src/stores/DataStore.ts [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+  CourseStatus,
+  UserDetails,
+  CourseState,
+  UserSettings,
+} from "../types/data-types";
 import { RootStore } from "./RootStore";
 
 export class DataStore {
   public userDetails: UserDetails = {} as UserDetails;
+  public userSettings: UserSettings = {} as UserSettings;
```
Comment (NissanOhana) [inline packages/sogrim-app/src/stores/DataStore.ts:14]: חכם

File: packages/sogrim-app/src/components/Banner/BannerCards/DegreeMainStatus.tsx [hunk 3]
```diff
@@ -1,60 +1,146 @@
...
+    _: React.SyntheticEvent,
+    computeInProgress: boolean
+  ) => {
+    setComputeInProgress(computeInProgress);
+    let newUserSettings =
+      updateComputeInProgressInUserSettings(computeInProgress);
+    mutate(newUserSettings);
+  };
+
   return showMainStatus ? (
```
Comment (NissanOhana) [inline packages/sogrim-app/src/components/Banner/BannerCards/DegreeMainStatus.tsx:106]: @liadaram1 
extract and break this component. its huge!

File: packages/sogrim-app/src/components/Banner/BannerCards/DegreeStatusBar.tsx [hunk 4]
```diff
@@ -1,9 +1,12 @@
 import * as React from "react";
 import LinearProgress, {
+  linearProgressClasses,
```
Comment (NissanOhana) [inline packages/sogrim-app/src/components/Banner/BannerCards/DegreeStatusBar.tsx:3]: noice

File: packages/sogrim-app/src/components/Banner/BannerCards/DegreeStatusBar.tsx [hunk 5]
```diff
@@ -1,9 +1,12 @@
 import * as React from "react";
```
Comment (NissanOhana) [inline packages/sogrim-app/src/components/Banner/BannerCards/DegreeStatusBar.tsx:1]: @liadaram1  @benny-n 
we need to break this component also

File: packages/sogrim-app/src/components/Banner/BannerCards/DegreeStatusBar.tsx [hunk 6]
```diff
@@ -14,18 +17,87 @@ const LinearProgressWithLabel = (
...
+}));
+
+const LinearProgressMultiColorWithLabel = (
+  props: LinearProgressProps & {
+    onlyCompleteProgress: number;
+    totalProgress: number;
+  }
+) => {
+  const theme = useTheme();
   return (
```
Comment (NissanOhana) [inline packages/sogrim-app/src/components/Banner/BannerCards/DegreeStatusBar.tsx:50]: @liadaram1  @benny-n 
best practice for tsx file - "only one return". i mean, only one component so we need to break this file.
each file - component.

File: packages/sogrim-app/src/components/Pages/SemestersPage/SemesterTable/CourseValidator.ts [hunk 7]
```diff
@@ -65,8 +65,8 @@ export const courseFromUserValidations = (
     };
   }
 
-  let newState = "לא הושלם";
   let newGrade = course.grade === "" ? undefined : course.grade;
+  let newState = newGrade ? "לא הושלם" : "בתהליך"; // The hebrew flips the conditions
```
Comment (NissanOhana) [inline packages/sogrim-app/src/components/Pages/SemestersPage/SemesterTable/CourseValidator.ts:69]: two validators files... :( 
@NissanOhana  - ביזיון!

**Review by NissanOhana — CHANGES_REQUESTED** [review]
Do we want to make a different color theme if the user clicks the toggle? @liadaram1  @benny-n @liadaram1

File: packages/sogrim-app/src/components/Banner/BannerCards/DegreeMainStatus.tsx (hunk 1 above)
Comment (liadaram1) [reply packages/sogrim-app/src/components/Banner/BannerCards/DegreeMainStatus.tsx:1]: I will open a new issue for that.
It's a change which shouldn't be done in the scope of this task.

Comment (liadaram1) [reply packages/sogrim-app/src/components/Banner/BannerCards/DegreeMainStatus.tsx:1]: I will open a new issue for that.
It's a change which shouldn't be done in the scope of this task.

**Comment by benny-n** [comment]
> Do we want to make a different color theme if the user clicks the toggle? @liadaram1 @benny-n @liadaram1

@NissanOhana not sure all of the color theme should change, but we do get a nice progress bar with 3 colors (complete, in-progress, not complete) when the user clicks the toggle

**Comment by NissanOhana** [comment]
> > Do we want to make a different color theme if the user clicks the toggle? @liadaram1 @benny-n @liadaram1
> 
> @NissanOhana not sure all of the color theme should change, but we do get a nice progress bar with 3 colors (complete, in-progress, not complete) when the user clicks the toggle

If so, yalla.

File: packages/sogrim-app/src/components/Banner/BannerCards/ComputeInProgressToggle.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
+import { Typography, Switch, Tooltip, Box } from "@mui/material";
```
Comment (NissanOhana) [inline packages/sogrim-app/src/components/Banner/BannerCards/ComputeInProgressToggle.tsx:1]: מקצוען

**Review by NissanOhana — APPROVED** [review]
Looking good.
all the remaining issues (spaghetti code in some component) are because of me 🤤 

@liadaram1 - if you checked all cases in the UI - we can merge :)

**Comment by NissanOhana** [comment]
IMO we can merge.
@benny-n  - want to PR new version stuff (bump version, version modal, etc...) 🕺

**Pull request merged**